client/src/components/Scroll/ScrollTop.js

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from "react";
+import { withRouter, useLocation } from "react-router-dom";
+
+function ScrollToTop() {
+  const { pathname } = useLocation();
+  const [flag, setFlag] = useState(true);
+
+  window.onbeforeunload = function () {
+    setFlag(false);
+  };
+
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [pathname, !flag]);
```

Comment:
I haven't seen this usage in dependencies, `!flag`, if flag changes then `useEffect `would cause a re-render. So just `flag` should be enough. Also, I tested this and haven't faced any issue.

Reply:
done